### ISSUE TITLE
generator: Support calls that return an error in attributes.

### ISSUE THREAD

**Issue by smit2909** [a-h/templ#492]
Sample code

```
import (
	"fmt"
)

func example() (string, error) {
	return "", fmt.Errorf("Not implemented yet.")
}

templ InvalidBlock() {
	<div boom={ example() }>SomeContent</div> //this complains,
}

templ ValidBlock() {
        <div>{ example() }</div> // this allows functions that return string, error
}
```

I see function expressions are supported in the actual value of the div tag, but attributes in a tag cannot support function expressions that return (string, error). I am wondering if this can be supported or is there a work around for this?

![image](https://github.com/a-h/templ/assets/79797111/25256fa6-e6dd-43b6-912a-7dea775259e2)

**Comment by joerdav** [comment]
Hi @smit2909 , I think you may have encountered an oversight when introducing error handling to templates. Currently no, HTML attributes cannot include error results.

**Comment by a-h** [comment]
Yes, I think it's a feature we'd want to add. Should be relatively straightforward to do, given the prior art of enabling it for `StringExpression`.

A change to the generator is required. For `StringExpression`, this is generated:

```go
var templ_7745c5c3_Var4 string
templ_7745c5c3_Var4, templ_7745c5c3_Err = templ.JoinStringErrs(doSomethingWithError("test"))
if templ_7745c5c3_Err != nil {
	return	templ.Error{Err: templ_7745c5c3_Err, FileName: ``, Line: 11, Col: 31}
}
```

For attribute expression, this is currently generated:

```go
_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString(doSomethingWithError("test")))
if templ_7745c5c3_Err != nil {
	return templ_7745c5c3_Err
}
```

The `JoinStringErrors` needs to be called, then the error check made, then output the results of the `templ.EscapeString` call.

**Comment by smit2909** [comment]
Oh, thanks for clarifying, appreciate marking this as an enhancement. I wonder if this might also need changes to templ lsp.

**Comment by a-h** [comment]
Because the templ LSP proxies go code through to gopls, it should be able to handle this without any changes.

**Comment by sirgallifrey** [comment]
Took a shot at the fix #574

**Comment by sirgallifrey** [comment]
@a-h this one can be closed due to #584

**Comment by a-h** [comment]
Thanks for that!
